services/web/client/source/class/osparc/store/Store.js

```diff
@@ -155,33 +155,24 @@ qx.Class.define("osparc.store.Store", {
      * @param {Boolean} reload ?
      */
     getServices: function(reload) {
-      if (!osparc.utils.Services.reloadingServices && (reload || Object.keys(osparc.utils.Services.servicesCached).length === 0)) {
-        osparc.utils.Services.reloadingServices = true;
-        osparc.data.Resources.get("servicesTodo", null, !reload)
-          .then(data => {
-            const allServices = data.concat(osparc.utils.Services.getBuiltInServices());
-            const filteredServices = osparc.utils.Services.filterOutUnavailableGroups(allServices);
-            const services = osparc.utils.Services.convertArrayToObject(filteredServices);
-            osparc.utils.Services.servicesToCache(services, true);
-            this.fireDataEvent("servicesRegistered", {
-              services,
-              fromServer: true
-            });
-          })
+      return new Promise((resolve, reject) => {
+        const allServices = osparc.utils.Services.getBuiltInServices();
+        const servicesPromise = osparc.data.Resources.get("servicesTodo", null, !reload);
+        const groupsPromise = osparc.data.Resources.get("groups", null, !reload);
+        Promise.all([servicesPromise, groupsPromise])
           .catch(err => {
```

Comment:
``catch`` should go after ``then``. Otherwise ``catch`` cannot catch errors happening inside ``then``, as it is before in the chain. But most importantly, when one of the promises inside ``Promise.all`` rejects, the ``then`` will execute after the ``catch``, and because the catch is not returning anything, ``values`` will be undefined and trying to access it as an array will fail